created a menu item for displaying app dependencies on ui

rohinish404 opened the pull request:
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission


## Have you discussed this change with the InvokeAI team?
- [X] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ ] No


## Description


## Related Tickets & Documents

<!--
For pull requests that relate or close an issue, please include them
below. 

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

- Related Issue #5223 
- Closes #5223 

## QA Instructions, Screenshots, Recordings

<!-- 
Please provide steps on how to test changes, any hardware or 
software specifications as well as any other pertinent information. 
-->

## Merge Plan

<!--
A merge plan describes how this PR should be handled after it is approved.

Example merge plans:
- "This PR can be merged when approved"
- "This must be squash-merged when approved"
- "DO NOT MERGE - I will rebase and tidy commits before merging"
- "#dev-chat on discord needs to be advised of this change when it is merged"

A merge plan is particularly important for large PRs or PRs that touch the
database in any way.
-->

## Added/updated tests?

- [ ] Yes
- [ ] No : _please replace this line with details on why tests
      have not been included_

## [optional] Are there any post deployment tasks we need to perform?

rohinish404 commented:
Feedbacks please
<img width="1084" alt="Screenshot 2024-01-12 at 6 07 01 PM" src="https://github.com/invoke-ai/InvokeAI/assets/92542124/adbadf39-e177-443b-802c-9925c6da0846">

hipsterusername reviewed (approved):
LGTM!

hipsterusername commented:
@rohinish404 - Lint tests are failing though

psychedelicious commented:
You'll need to add translations

rohinish404 commented:
> You'll need to add translations

Added!

rohinish404 commented:
@psychedelicious @hipsterusername  should i add links to yt and twitter also?

hipsterusername commented:
With the current styling as text links probably best to leave as is

rohinish404 commented:
> With the current styling as text links probably best to leave as is

Okay..then it is ready to be merged! All tests passed!

psychedelicious reviewed (changes requested):
Blocking merge until I've reviewed

psychedelicious commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,140 @@
+  const { t } = useTranslation();
+  const { data: appDeps } = useGetAppDepsQuery();
+  const { data: appVersion } = useGetAppVersionQuery();
+  const appDeps_arr = map(toPairs(appDeps), (d) => fromPairs([d]));
```
This is a bit convoluted. Let's leverage RTK Query to make this simpler:

```ts
  const { deps } = useGetAppDepsQuery(undefined, {
    selectFromResult: ({ data }) => ({
      deps: data ? map(data, (version, name) => ({ name, version })) : [],
    }),
  });
```

Then you can map over `deps` easily:
```ts
deps.map(({ name, version }, i) => (
  // ... do something with name and version
))
```

psychedelicious commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,140 @@
+                      {t('common.aboutHeading')}
+                    </InvHeading>
+                    <InvText fontSize="sm">{t('common.aboutDesc')}</InvText>
+                    <Link href={websiteLink} fontSize="sm">
+                      {t('common.websiteLink')}
+                    </Link>
```
Use `websiteLink` as the text. There's no need to use translation here.

Also, this is an external link.

psychedelicious commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
The layout and implementation can be cleaned up and we can provide a better user experience.

PR design:

https://github.com/invoke-ai/InvokeAI/assets/4822129/bcfeec29-16ed-4760-86da-519c9b618d82

Desired design:

https://github.com/invoke-ai/InvokeAI/assets/4822129/5021327c-d69a-4315-a627-37c8288c3078

Tips:
- You may need to rethink the layout to accomplish the design.
- The usage of grid here is a bit awkward. There's no need for it where it's used. The table is the thing that makes sense to be grid. 
- That said, it may be easier to just use flex for all the components than fiddle around with grid or table.
- You can use `position: sticky` to make the title stick at the top.
- You may need to wrap `ScrollableContent` in another element to get it to work and give it padding/bg. 
- The modal dimensions should be set to fit the content but use the 80vh max.

psychedelicious reviewed (changes requested):
Thanks for working on this @rohinish404!

I've requested a few changes.

I've also provided a desired design in a video. If you're up for the challenge, you can implement the design. If you are tapped out on this PR, no worries, I can update it.

rohinish404 commented:
> Thanks for working on this @rohinish404!
> 
> I've requested a few changes.
> 
> I've also provided a desired design in a video. If you're up for the challenge, you can implement the design. If you are tapped out on this PR, no worries, I can update it.

I'll start working on these! Thanks for the feedbacks! :)

rohinish404 commented on invokeai/frontend/web/public/locales/en.json:
```diff
@@ -199,7 +204,9 @@
-        "unsaved": "Unsaved"
+        "unsaved": "Unsaved",
+        "v": "v",
+        "websiteLink": "www.invoke.com"
```
Done with latest commit!

rohinish404 commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,140 @@
+  const { data: appDeps } = useGetAppDepsQuery();
+  const { data: appVersion } = useGetAppVersionQuery();
+  const appDeps_arr = map(toPairs(appDeps), (d) => fromPairs([d]));
+  const ref = useRef(null);
```
Done with latest commit!

rohinish404 commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,140 @@
+  const ref = useRef(null);
+  const githubLink = 'http://github.com/invoke-ai/InvokeAI';
+  const discordLink = 'https://discord.gg/ZmtBAhwWhy';
+  const websiteLink = 'https://www.invoke.com/';
```
Haven't changed in SettingsMenu component rn. Will change it with next commit!

rohinish404 commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,140 @@
+                      h="80px"
+                      minW="24px"
+                      minH="24px"
+                      userSelect="none"
```
Done with latest commit!

rohinish404 commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,140 @@
+                    <InvText>
+                      {t('common.v')}
+                      {appVersion?.version}
+                    </InvText>
```
Done with latest commit!

rohinish404 commented:
All requested changes done! Now further feedback required!

psychedelicious commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,133 @@
+                      justifyContent="space-between"
+                    >
+                      <InvText>{name}</InvText>
+                      <InvText>{version ? version : 'Not Installed'}</InvText>
```
Need a translation for this

psychedelicious commented on invokeai/frontend/web/src/features/system/components/AboutModal/AboutModal.tsx:
```diff
@@ -0,0 +1,133 @@
+      })}
+      <InvModal isOpen={isOpen} onClose={onClose} isCentered size="2xl">
+        <InvModalOverlay />
+        <InvModalContent maxH="80vh" h="80vh">
```
The 80vh height is a bit too much - it means the modal can be way too tall.

Let's set the height to a value that shows all of the dependencies without scrolling, leaving the maxH where it's at.

rohinish404 replied:
My screen size is 13 inch so its looking fine on my screen
<img width="1084" alt="Screenshot 2024-01-12 at 6 07 01 PM" src="https://github.com/invoke-ai/InvokeAI/assets/92542124/888e4bbe-f6dd-4165-9bca-e17af11bbd50">

rohinish404 replied:
any height you suggest to go with?

psychedelicious reviewed (changes requested):
That's looking great now! Two small changes requested.

rohinish404 commented:
@psychedelicious couple of pointers on the latest commit - 

- changed the height of both components to 50 as we discussed in discord.
- replaced flex with grid to make the space between deps and versions less (basically to match the example you gave).
- replaced the margin top with justify content as you suggested.

Millu commented:
Got this error while trying to test: 

 ``` 
at InvokeTabs (http://localhost:5173/src/features/ui/components/InvokeTabs.tsx:152:28)
    at div
    at http://localhost:5173/node_modules/.vite/deps/chunk-JJ4RZU4H.js?v=a76f4282:1396:45
    at ChakraComponent (http://localhost:5173/node_modules/.vite/deps/chunk-7IPMFKXC.js?v=a76f4282:5539:35)
    at ErrorBoundary (http://localhost:5173/node_modules/.vite/deps/react-error-boundary.js?v=a76f4282:18:5)
    at App (http://localhost:5173/src/app/components/App.tsx:36:16)
    at DndContext2 (http://localhost:5173/node_modules/.vite/deps/@dnd-kit_core.js?v=a76f4282:2494:5)
    at DndContextTypesafe
    at AppDndContext (http://localhost:5173/src/features/dnd/components/AppDndContext.tsx:22:49)
    at EnvironmentProvider (http://localhost:5173/node_modules/.vite/deps/chunk-PUI7XM66.js?v=a76f4282:68:11)
    at ColorModeProvider (http://localhost:5173/node_modules/.vite/deps/chunk-7IPMFKXC.js?v=a76f4282:328:5)
    at ThemeProvider2 (http://localhost:5173/node_modules/.vite/deps/chunk-JJ4RZU4H.js?v=a76f4282:1443:22)
    at ThemeProvider2 (http://localhost:5173/node_modules/.vite/deps/chunk-7IPMFKXC.js?v=a76f4282:5616:11)
    at ChakraProvider (http://localhost:5173/node_modules/.vite/deps/@chakra-ui_react.js?v=a76f4282:579:5)
    at ChakraProvider22 (http://localhost:5173/node_modules/.vite/deps/@chakra-ui_react.js?v=a76f4282:1344:5)
    at ThemeLocaleProvider (http://localhost:5173/src/app/components/ThemeLocaleProvider.tsx:18:32)
    at Suspense
    at Provider (http://localhost:5173/node_modules/.vite/deps/chunk-UFRHBOCB.js?v=a76f4282:1089:3)
    at InvokeAIUI (http://localhost:5173/src/app/components/InvokeAIUI.tsx:33:23)
```

Full log: 
[localhost-1705429568363.log](https://github.com/invoke-ai/InvokeAI/files/13954398/localhost-1705429568363.log)

psychedelicious reviewed (approved):
Thanks! I made a couple very minor changes.

rohinish404 commented:
> Thanks! I made a couple very minor changes.

Awesome! Thanks for the merge!🙂
